### DISCUSSION TITLE
Using USEclipboard will refresh render?

### DISCUSSION THREAD

**Discussion by kringt06** [chakra-ui/chakra-ui#4126] [Q&A] [answered]
Using USEclipboard will refresh render twice?

```tsx
 const { isOpen, onOpen, onClose } = useDisclosure()
 console.log('render')
```
will print ->
render
render

**Comment by anubra266** [comment]
Could you provide a more descriptive code, there's nothing like `useClipboard` in this code. Thanks

**Reply by kringt06** [reply]
```tsx
const App = () => {
    const { isOpen, onOpen, onClose } = useDisclosure() 
    console.log('render')
    return <Box>...</Box>
}
```

package.json =>

"dependencies": {
    "@apollo/client": "^3.3.19",
    "@chakra-ui/icons": "^1.0.13",
    "@chakra-ui/react": "^1.6.2",
    "@chakra-ui/theme-tools": "1.1.7",
    "@emotion/react": "11.4.0",
    "@emotion/styled": "11.3.0",
    "cross-env": "^7.0.3",
    "dayjs": "^1.10.5",
    "formik": "^2.2.8",
    "framer-motion": "4.1.17",
    "graphql": "^15.5.0",
    "js-cookie": "^2.2.1",
    "mathjs": "^9.4.1",
    "mobx": "^6.3.2",
    "mobx-react": "^7.2.0",
    "next": "^10.2.3",
    "next-i18next": "^8.3.0",
    "next-images": "^1.7.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-infinite-scroll-component": "^6.1.0",
    "react-remove-scroll": "^2.4.2",
    "react-spring": "^9.1.2",
    "recharts": "^2.0.9",
    "sass": "^1.34.0"
  },

**Reply by kringt06** [reply]
use Nextjs ssr

**Comment by anubra266** [comment, marked as answer]
It's not a Chakra Bug, It's the way react works. It won't happen in production.
 You should read this
https://mariosfakiolas.com/blog/my-react-components-render-twice-and-drive-me-crazy/
